pkg/agent/flowexporter/connections/connections.go

```diff
@@ -178,24 +178,57 @@ func (cs *connectionStore) addOrUpdateConn(conn *flowexporter.Connection) {
 			egressOfID := binary.LittleEndian.Uint32(conn.Labels[4:8])
 			if ingressOfID != 0 {
 				policy := cs.networkPolicyQuerier.GetNetworkPolicyByRuleFlowID(ingressOfID)
+				rule := cs.networkPolicyQuerier.GetRuleByFlowID(ingressOfID)
 				if policy == nil {
 					// This should not happen because the rule flow ID to rule mapping is
 					// preserved for max(5s, flowPollInterval) even after the rule deletion.
 					klog.Warningf("Cannot find NetworkPolicy that has rule with ingressOfID %v", ingressOfID)
 				} else {
 					conn.IngressNetworkPolicyName = policy.Name
 					conn.IngressNetworkPolicyNamespace = policy.Namespace
+					conn.IngressNetworkPolicyUID = string(policy.UID)
+					switch policy.Type {
+					case "K8sNetworkPolicy":
+						conn.IngressNetworkPolicyType = 1
+					case "AntreaNetworkPolicy":
+						conn.IngressNetworkPolicyType = 2
+					case "AntreaClusterNetworkPolicy":
+						conn.IngressNetworkPolicyType = 3
+					}
+					conn.IngressNetworkPolicyRuleName = rule.Name
+					// TODO: tier name, tier priority, policy priority to be added when support available
+					if policy.Type == "K8sNetworkPolicy" {
+						conn.IngressNetworkPolicyRulePriority = -1
+					} else {
+						conn.IngressNetworkPolicyRulePriority = int32(*rule.Priority)
+					}
 				}
 			}
 			if egressOfID != 0 {
 				policy := cs.networkPolicyQuerier.GetNetworkPolicyByRuleFlowID(egressOfID)
+				rule := cs.networkPolicyQuerier.GetRuleByFlowID(egressOfID)
 				if policy == nil {
 					// This should not happen because the rule flow ID to rule mapping is
 					// preserved for max(5s, flowPollInterval) even after the rule deletion.
 					klog.Warningf("Cannot find NetworkPolicy that has rule with egressOfID %v", egressOfID)
 				} else {
 					conn.EgressNetworkPolicyName = policy.Name
 					conn.EgressNetworkPolicyNamespace = policy.Namespace
+					conn.EgressNetworkPolicyUID = string(policy.UID)
+					switch policy.Type {
+					case "K8sNetworkPolicy":
```

Comment:
It would be better to use const from definitions like`pkg/apis/controlplane/types.go` instead of string here.